cwl/echo.cwl

```diff
@@ -1,21 +1,66 @@
-arguments:
-- $(inputs.message)
-baseCommand:
-- echo
-- -n
+#!/usr/bin/env cwl-runner
+
 class: CommandLineTool
 cwlVersion: v1.0
+
+label: Print given message using `echo` and capture output
+
+doc: >
+  The `echo` command-line utility is used to print a message,
+  which is captured from standard output.
+
+  Visit the Linux manual page for more information:
+    https://www.freebsd.org/cgi/man.cgi?query=echo
+
+requirements:
+  - class: ShellCommandRequirement
+
 hints:
   DockerRequirement:
     dockerPull: sagebionetworks/dockstore-tool-template:0.0.4
+
+baseCommand:
+  - echo
+
 inputs:
+
+  extra_args:
+    type: string?
+    label: Additional command-line (usually optional) arguments
+    inputBinding:
+      position: 0
+      shellQuote: false
+
   message:
     type: string
+    inputBinding:
+      position: 1
+
 outputs:
-  out:
+
+  output_file:
+    type: File
+    format: edam:format_1964  # Plain text
+    outputBinding:
+      glob: out.txt
+  
+  output_contents:
+    type: string
     outputBinding:
       glob: out.txt
       loadContents: true
       outputEval: $(self[0].contents)
-    type: string
+
 stdout: out.txt
+
+$namespaces:
+  s: https://schema.org/
+  edam: http://edamontology.org/
+
+s:author:
+  - class: s:Person
+    s:identifier: https://orcid.org/0000-0002-4621-1589
+    s:email: bruno.grande@sagebase.org
+    s:name: Bruno Grande
```

Comment:
Neither you nor I are really the original author -- what I wrote was based upon this [cwl](https://github.com/common-workflow-language/cwltool/blob/main/tests/echo.cwl) from the cwltool repository. 

I deliberately did not add annotations, labels, or doc because this example was never intended to illustrate CWL basics but rather as the _simplest_ possible example to illustrate project organization and CICD concepts. However, what you've added is valuable because anyone using the template may not have studied CWL yet and could benefit from an example that includes all the trimmings.


My suggestion here is that you leave the original file as written. Create a secondary one called echo_extra_args.cwl. In that one you can make your changes. If you follow that path, please add a line or two to the README indicating that `echo.cwl` is the simple example, but we've added `echo_extra_args.cwl` to illustrate recommended best practices.

Reply:
Thank you for providing the historical context for this repo. I know understand why the `echo.cwl` file was barebones. I'll create a separate file containing my more detailed example.